core/src/env_platform/db_handler.bal

```diff
@@ -73,10 +50,44 @@ function saveApplication(TreeRemovalForm form) returns error? {
                     "district": form.district,
                     "nameOfTheLand": form.nameOfTheLand,
                     "planNumber": form.planNumber,
-                    "area": locations,
-                    "treeInformation": treeInformation
+                    "area": getAreaJsonArray(form.area),
+                    "treeInformation": getTreeInformationJsonArray(form.treeInformation)
                 }
             ]
     };
     return applicationCollection->insert(application);
 }
+
+# The `deleteApplication` function will delete application drafts only which status is save.
+# 
+# + applicationId - The Id of the application which must be deleted
+# + return - This function will return null if application is deleted from the database or else return mongodb:DatabaseError
+# array index out of bound if there are no application with the specific application Id.
+function deleteApplication(string applicationId) returns boolean|error {
+
+    string applicationType = check getApplicationTypeByApplicationId(applicationId);
```

Comment:
```suggestion
    string applicationStatus = check getApplicationTypeByApplicationId(applicationId);
```
I also suggest changing **applicationType** field to **applicationStatus** to make it more relevant to the value it is returning. It must be changed in all other places in the code if that is the case, including method names